core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -393,7 +393,7 @@ final class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: Stri
    * A background reaper to expire delayed operations that have timed out
    */
   private class ExpiredOperationReaper extends ShutdownableThread(
-    "ExpirationReaper-%d".format(brokerId),
+    "ExpirationReaper-%d-%s".format(brokerId, purgatoryName),
```

Comment:
Just double checking if any of the broker side metrics would depend on this? If yes it may be a backward compatibility broken change. @ijuma @junrao ?

EDIT: made a pass and seems there is none.